numpy/fft/_pocketfft.c

```diff
@@ -2228,17 +2237,26 @@ execute_complex(PyObject *a1, int is_forward, double fct)
 }
 
 static PyObject *
-execute_real_forward(PyObject *a1, double fct)
+execute_real_forward(PyObject *a1, double fct, PyObject *out)
 {
     rfft_plan plan=NULL;
     int fail = 0;
     npy_intp tdim[NPY_MAXDIMS];
 
-    PyArrayObject *data = (PyArrayObject *)PyArray_FromAny(a1,
+    PyArrayObject *data;
+    if(out == Py_None) {
+      data = (PyArrayObject *)PyArray_FromAny(a1,
             PyArray_DescrFromType(NPY_DOUBLE), 1, 0,
             NPY_ARRAY_DEFAULT | NPY_ARRAY_ENSUREARRAY | NPY_ARRAY_FORCECAST,
             NULL);
-    if (!data) return NULL;
+      if (!data) return NULL;
+    }
+    else {
+      data = (PyArrayObject*)PyArray_EnsureArray(out);
+      if(!data) return NULL;
+      if(PyArray_CopyObject(data, a1) == -1) return NULL;
```

Comment:
I don't think this is right for `fftr` since `out` does not have the same shape and dtype as the input. Instead, this should be around line 2267 (`PyArrayObject *ret = (PyArrayObject *)PyArray_Empty(ndim, tdim, PyArray_DescrFromType(NPY_CDOUBLE), 0);`